### PR TITLE
Basic ElasticSearch VectorStore

### PR DESCRIPTION
**Pull request by cdiddy77**
Add support for Elasticsearch as a vector store.

Partially addresses #771 

This is the first step for Elasticsearch support, a basic vector store, to be followed by embeddings and Knn search.

### CODE REVIEW

**Comment by ezimuel** [comment]
Hi @cdiddy77 I didn't notice you was working on the Elasticsearch vectorstore. I did the same in #1784 including documentation and examples. Please have a look and let me know, thanks.

**Comment by cdiddy77** [comment]
> Hi @cdiddy77 I didn't notice you was working on the Elasticsearch vectorstore. I did the same in #1784 including documentation and examples. Please have a look and let me know, thanks.

Looks good. I added a few comments.

File: langchain/src/vectorstores/elasticsearch.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+interface VectorDocument {
+  vector: number[];
+  text: string;
+  metadata?: object;
+}
+
+/**
+ * Wrapper around Elasticsearch as a vector database.
+ */
+export class ElasticSearchStore extends VectorStore {
```
Comment (jacoblee93) [inline langchain/src/vectorstores/elasticsearch.ts:27]: In Python this seems to be called `ElasticVectorSearch`: https://python.langchain.com/docs/modules/data_connection/vectorstores/integrations/elasticsearch

Not sure if you prefer the synchronization or if this is a better name now?

File: langchain/src/vectorstores/elasticsearch.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            params: { query_vector: query },
+          },
+        },
+      },
+      size: k,
+    };
+    const response = await this.client.search<VectorDocument>(searchQuery);
+    return response.hits.hits.map((hit) => [
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      new Document<Record<string, any>>({
```
Comment (jacoblee93) [inline langchain/src/vectorstores/elasticsearch.ts:123]: The default type for `Document` should cover this, no need to explicitly state it and require TS suppression.

**Pull request closed without merging**